### PR TITLE
10467 -parameter override ui

### PR DESCRIPTION
**Pull request by unorthodoxgeek**

### CODE REVIEW

File: app/assets/javascripts/host_edit.js [hunk 1]
```diff
@@ -2,6 +2,47 @@ $(document).on('ContentLoad', function(){onHostEditLoad()});
 $(document).on('AddedClass', function(event, link){load_puppet_class_parameters(link)});
 $(document).on('click', '#params-tab', function() { resizeTextareas($('#params')); });
 
+LookupValueController = function(item) {
```
Comment (waldenraines) [inline app/assets/javascripts/host_edit.js:5]: Should this "controller" be separated out into it's own file?

File: app/assets/javascripts/host_edit.js [hunk 2]
```diff
@@ -2,6 +2,47 @@ $(document).on('ContentLoad', function(){onHostEditLoad()});
...
+  self.removeOverride = function() {
+    // remove/hide the overridden field
+    self.element.removeClass("overridden");
+    self.overrideTR.removeClass("visible");
+    self.destroyer.val("true");
+    return false;
+  }
+
+  self.overridden = function() {
+    self.element.hasClass("overridden");
```
Comment (waldenraines) [inline app/assets/javascripts/host_edit.js]: Did you mean to return this value?

File: app/assets/javascripts/host_edit.js [hunk 3]
```diff
@@ -2,6 +2,47 @@ $(document).on('ContentLoad', function(){onHostEditLoad()});
...
+
+  self.removeOverride = function() {
+    // remove/hide the overridden field
+    self.element.removeClass("overridden");
+    self.overrideTR.removeClass("visible");
+    self.destroyer.val("true");
+    return false;
+  }
+
+  self.overridden = function() {
```
Comment (waldenraines) [inline app/assets/javascripts/host_edit.js]: Perhaps `isOverridden` is a better name?

File: app/assets/javascripts/host_edit.js [hunk 4]
```diff
@@ -2,6 +2,47 @@ $(document).on('ContentLoad', function(){onHostEditLoad()});
...
+
+  self.addEventListeners = function() {
+    self.element.find(".override_button").click(self.override);
+    self.overrideTR.find(".remove_override_button").click(self.removeOverride);
+  }
+
+  self.override = function() {
+    //create the overridden field
+    self.element.addClass("overridden");
+    self.overrideTR.addClass("visible");
```
Comment (waldenraines) [inline app/assets/javascripts/host_edit.js]: `self.overrideTR.show();`?

File: app/assets/javascripts/host_edit.js [hunk 5]
```diff
@@ -2,6 +2,47 @@ $(document).on('ContentLoad', function(){onHostEditLoad()});
...
+    self.destroyer.val("false");
+    return false;
+  }
+
+  self.removeOverride = function() {
+    // remove/hide the overridden field
+    self.element.removeClass("overridden");
+    self.overrideTR.removeClass("visible");
```
Comment (waldenraines) [inline app/assets/javascripts/host_edit.js]: `self.overrideTR.hide();`?

File: app/assets/javascripts/host_edit.js (hunk 1 above)
Comment (unorthodoxgeek) [inline app/assets/javascripts/host_edit.js:5]: naa, no need for the extra GET request

File: app/assets/javascripts/host_edit.js (hunk 2 above)
Comment (unorthodoxgeek) [inline app/assets/javascripts/host_edit.js]: true, I think in Ruby

File: app/assets/javascripts/host_edit.js (hunk 4 above)
Comment (unorthodoxgeek) [inline app/assets/javascripts/host_edit.js]: I don't like manipulating inline CSS, even if it's done using JS, I prefer the class addition...

File: app/assets/javascripts/host_edit.js (hunk 1 above)
Comment (waldenraines) [inline app/assets/javascripts/host_edit.js:5]: Extra get request?  Aren't these JS files all combined down into one minified file?  The extra GET would only be present in development mode.  Splitting this out into its own file would make it easier to find in the future.

Comment (unorthodoxgeek) [inline app/assets/javascripts/host_edit.js:5]: nope, rails doesn't do that. it minifies the files, but doesn't unify them.

Comment (waldenraines) [inline app/assets/javascripts/host_edit.js:5]: You sure about that?  In production mode it looks like there is a single script and style GET request as expected.

File: app/assets/javascripts/host_edit.js (hunk 4 above)
Comment (waldenraines) [inline app/assets/javascripts/host_edit.js]: :+1: 

That's fine, I don't feel strongly enough about it to hold up these changes.

File: app/assets/javascripts/host_edit.js [hunk 6]
```diff
@@ -2,6 +2,43 @@ $(document).on('ContentLoad', function(){onHostEditLoad()});
 $(document).on('AddedClass', function(event, link){load_puppet_class_parameters(link)});
 $(document).on('click', '#params-tab', function() { resizeTextareas($('#params')); });
 
+LookupValueController = function(item) {
+  var self = this; //because JS' this confuses me
```
Comment (tbrisker) [inline app/assets/javascripts/host_edit.js:6]: I understand you like ruby better, but trying to force JS look like ruby isn't a good idea in my opinion. 
After you finish working on this code other developers will need to work with it and they will be looking for JS conventions (e.g. `prototype`) when working on a JS file.

Comment (unorthodoxgeek) [inline app/assets/javascripts/host_edit.js:6]: You are welcome to refactor this :)

Comment (unorthodoxgeek) [inline app/assets/javascripts/host_edit.js:6]: for instance, the this variable gets replaced in the callbacks, so I don't see how I can put the callbacks in prototype. I may not know JS well enough though...

**Pull request closed without merging**